src/lib/linked-list.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const Node = require('./node');
+const NodeTwo = require('./node-two');
+
+module.exports = class LinkedList {
+  constructor() {
+    this.head = null;
+  }
+
+  // Big O: O(1)
+  insertAtHead(value) {
+    const node = new Node(value);
+
+    node.next = this.head;
+    this.head = node;
+    return this;
+  }
+
+  // Big O: O(n)
+  insertAtEnd(value) {
+    const node = new Node(value);
+
+    if (!this.head) {
+      this.head = node;
+      return this;
+    }
+
+    let currentNode = this.head;
+    while (currentNode.next) {
+      currentNode = currentNode.next;
+    }
+
+    currentNode.next = node;
+    return this;
+  }
+
+  // Big O: O(n)
+  find(value) {
+    if (!this.head) {
+      throw new Error('__ERROR__ The list is empty');
+    }
+
+    if (this.head.value === value) {
+      return this.head;
+    }
+
+    let currentNode = this.head;
+    while (currentNode.next) {
+      if (currentNode.next.value === value) {
+        return currentNode.next;
+      }
+      currentNode = currentNode.next;
+    }
+    return null;
+  }
+
+  // Big O: O(n)
+  remove(value) {
+    if (!this.head) {
+      throw new Error('__ERROR__ The list is empty');
+    }
+
+    if (this.head.value === value) {
+      this.head = this.head.next;
+    } else {
+      let previousNode = this.head;
+      let currentNode = previousNode.next;
+      while (currentNode) {
+        if (currentNode.data === value) {
+          previousNode.next = currentNode.next;
+          currentNode = currentNode.next;
+          break;
+        } else {
+          previousNode = currentNode;
+          currentNode = currentNode.next;
+        }
+      }
+    }
+    return this;
+  }
+
+  // Big O: O(n)
+  double(value) {
```

Comment:
Good start, but think about what an array.map function does. It takes in a callback and iterates through each element of the array, and performs the callback on it, returning the results in a new array, right? The map function can do anything to the element, not necessarily just double it.

A written out array.map function would look something like:

Array.prototype.map = function(callback) {
  let result = [];
  for (let i in arr) result.push(callback(arr[i])); 
  return result;
}

Then you can invoke the map function by doing something like:
arr.map(element => element *= 2); 
And you specify directly in the callback what you want the map function to be doing. 

Try to do the same but for a linked list, and also give a try at the reduce method.